src/plugins/discover/public/application/main/components/no_results/no_results_suggestions/no_results_suggestions.tsx

```diff
@@ -132,28 +140,35 @@ export const NoResultsSuggestions: React.FC<NoResultsSuggestionProps> = ({
       }
       body={body}
       actions={
-        <div
-          css={css`
-            min-block-size: ${euiTheme.size.xxl};
-          `}
-        >
-          {typeof occurrencesRange === 'undefined' ? (
-            !isTextBasedQuery(query) && <EuiLoadingSpinner />
-          ) : canExtendTimeRange ? (
-            <EuiButton
-              color="primary"
-              fill
-              onClick={extendTimeRange}
-              isLoading={isExtending}
-              data-test-subj="discoverNoResultsViewAllMatches"
-            >
-              <FormattedMessage
-                id="discover.noResults.suggestion.viewAllMatchesButtonText"
-                defaultMessage="View all matches"
-              />
-            </EuiButton>
-          ) : null}
-        </div>
+        !isTextBasedQuery(query) ? (
+          <div
+            css={css`
+              min-block-size: ${euiTheme.size.xxl};
+            `}
+          >
+            {timeRangeExtendingStatus === TimeRangeExtendingStatus.notFound ? (
+              <EuiText color="subdued">
+                <FormattedMessage
+                  id="discover.noResults.suggestion.searchAllMatchesButtonGivesNoResultsText"
+                  defaultMessage="Expanding the time range gives no results too."
+                />
+              </EuiText>
+            ) : (
+              <EuiButton
+                color="primary"
+                fill
+                onClick={extendTimeRange}
+                isLoading={timeRangeExtendingStatus === TimeRangeExtendingStatus.loading}
+                data-test-subj="discoverNoResultsViewAllMatches"
+              >
+                <FormattedMessage
+                  id="discover.noResults.suggestion.searchAllMatchesButtonText"
+                  defaultMessage="Search entire time range"
```

Comment:
@kertal I saw that you suggested using "Search all matches". But when looking at the whole screen, I think it would be very clear (what happens when pressed) if we use something like "Search entire time range" instead.

<img width="870" alt="Screenshot 2024-04-25 at 16 21 57" src="https://github.com/elastic/kibana/assets/1415710/263bacc6-7c81-43c2-a18a-2ff3b97e195b">

wdyt?

cc @amyjtechwriter

Reply:
for me this makes lots of sense (I just wrote what came to my mind first 👍 )

Reply:
since we move to a different wording here, we should consider to just search for the earliest and latest timestamp, ignoring the other query part. It's because depending on the query and context, it can be expensive. for a random query like {"query":"fsdfssdfw332"} every field on every shard needs to be searched


And it seems the timeout we can configure in Kibana is ignored. We could consider to use this timeout, or even a shorter one.

Reply:
About the timeout, this could be done in a separate PR, and back ported, for me this sounds like a fix (but I didn't investigate how this works with the timeout currently, I was just mentioned in an email, that this request is triggering an async request without timeout)

Reply:
> we should consider to just search for the earliest and latest timestamp, ignoring the other query part

I think keeping the query and filters is what makes this feature helpful so users don't have to guess the time range where such documents occur.

Reply:
> that this request is triggering an async request without timeout

@lukasolson How does the timeout work? Is it applied to all `data.search.search` by default or not?

I think if users refresh the page in between or close the browser tab then requests would keep running.

Reply:
"Search entire time range" seems decent to me. Maybe "Search full time range" would be a bit more succinct, or even something like "Search all documents"? Another good question for @amyjtechwriter 😁

Reply:
@davismcphee and @jughosta I really like 'Search entire time range' because it's very clear, great stuff!